src/models/environment.ts

```diff
@@ -1,24 +1,63 @@
 import { Data, type Require as Enforced } from "dataclass";
 import * as vscode from "vscode";
-import { CCLOUD_CONNECTION_ID, IconNames } from "../constants";
+import { CCLOUD_CONNECTION_ID, IconNames, LOCAL_CONNECTION_ID } from "../constants";
 import { CustomMarkdownString } from "./main";
 
+/**
+ * Base class between local and ccloud environments, and possibly
+ * direct connections in the future.
+ *
+ * An environment is a distinct collection of resources, namely
+ * Kafka clusters, a possible Schema Registry, and perhaps more
+ * things in the future such as Flink clusters.
+ *
+ */
+export abstract class Environment extends Data {
+  abstract readonly connectionId: string;
+  abstract readonly isLocal: boolean;
+  abstract readonly isCCloud: boolean;
+
+  id!: Enforced<string>;
+  name!: Enforced<string>;
+
+  /**
+   * Has at least one Kafka cluster or Schema Registry.
+   *
+   * CCloud environemts may have neither (yet), but we still want to show
+   * them in the tree.
+   */
+  abstract hasClusters: boolean;
+
+  // It would seem natural for the Environment to have a list of clusters,
+  // optional schema registry, and so on, but this hasn't grown to be
+  // the case yet.
```

Comment:
I've gone back and forth with this a few times with our modeling, and for the time being I like the idea of using the extension's (workspace) state as a centralized place to find all associated data (via the ResourceManager). Could be worth trying to link these resource types together to see if it's even easier to traverse, though. (E.g. look up topics based on a schema definition.)